services/ui-src/src/components/app/App.tsx

```diff
@@ -12,7 +14,9 @@ export const App = () => {
         <Flex sx={sx.appLayout}>
           <Header handleLogout={logout} />
           <Container sx={sx.appContainer} data-testid="app-container">
-            <AppRoutes userRole={user.userRole} />
+            <ErrorBoundary FallbackComponent={Error}>
+              <AppRoutes userRole={user?.userRole} />
+            </ErrorBoundary>
```

Comment:
Wrapping all the app routes with an error boundary, which is where errors are most likely to occur. Wrapping it here allows us to make the nice styled Error component as the fallback.